### ISSUE TITLE
False postive for `MaxChainedCallsOnSameLine` rule

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#5696]
## Expected Behavior
```
fun Int.identity(value: Int) = this + value
val x = 1.identity(
    1
).plus(1).plus(1)
```
above code should not be reported by the rule for the threshold of `maxChainedCalls` to be 3
## Observed Behavior
Rule reports this a warning

## Steps to Reproduce
Failing TC that should pass
```kotlin
@Test
fun `does not count a multiline fun call as chained calls`() {
    val rule = MaxChainedCallsOnSameLine(TestConfig(mapOf("maxChainedCalls" to 3)))
    val code = """
        fun Int.identity(value: Int) = this + value
        val x = 1.identity(
            1
        ).plus(1).plus(1)
    """.trimIndent()
    assertThat(rule.compileAndLintWithContext(env, code)).isEmpty()
}
```

## Your Environment

* Version of detekt used: 1.22.0

**Comment by BraisGabin** [comment]
Closing in favor of #5598
